paddle/fluid/framework/ir/graph_pattern_detector.cc

```diff
@@ -1796,80 +1796,22 @@ PDNode *patterns::Conv::operator()() {
   return output_var;
 }
 
-PDNode *patterns::Transpose::operator()() {
+PDNode *patterns::Immutable::operator()(const std::string immutable_type) {
   auto prev_op = pattern->NewNode(prev_op_repr())->assert_is_op();
 
-  auto transpose_op =
-      pattern->NewNode(transpose_op_repr())->assert_is_op("transpose2");
+  auto immutable_op =
+      pattern->NewNode(immutable_op_repr())->assert_is_op(immutable_type);
 
-  auto transpose_in = pattern->NewNode(transpose_in_repr())
+  auto immutable_in = pattern->NewNode(immutable_in_repr())
                           ->AsInput()
-                          ->assert_is_op_input("transpose2");
-  auto transpose_out = pattern->NewNode(transpose_out_repr())
+                          ->assert_is_op_input(immutable_type);
```

Comment:
I'm wondering if we should add name of the input "X" and for slice op "Input", because some of them have many inputs. It can be validate in cpu_quantize_pass or in graph_pattern_detector by passing input_name.

Reply:
That makes sense, I added your suggestion.